x-pack/plugins/endpoint/public/embeddables/resolver/lib/matrix3.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Matrix3 } from '../types';
+
+/**
+ * Return a new matrix which is the product of the first and second matrix.
+ */
+export function multiply(
+  [a11, a12, a13, a21, a22, a23, a31, a32, a33]: Matrix3,
+  [b11, b12, b13, b21, b22, b23, b31, b32, b33]: Matrix3
+): Matrix3 {
+  const s11 = a11 * b11 + a12 * b21 + a13 * b31;
+  const s12 = a11 * b12 + a12 * b22 + a13 * b32;
+  const s13 = a11 * b13 + a12 * b23 + a13 * b33;
+
+  const s21 = a21 * b11 + a22 * b21 + a23 * b31;
+  const s22 = a21 * b12 + a22 * b22 + a23 * b32;
+  const s23 = a21 * b13 + a22 * b23 + a23 * b33;
+
+  const s31 = a31 * b11 + a32 * b21 + a33 * b31;
+  const s32 = a31 * b12 + a32 * b22 + a33 * b32;
+  const s33 = a31 * b13 + a32 * b23 + a33 * b33;
+
+  // prettier-ignore
+  return [
+    s11, s12, s13,
+    s21, s22, s23,
+    s31, s32, s33,
+  ];
+}
+
+/**
+ * Return a new matrix which is the sum of the two passed in.
+ */
+export function add(
```

Comment:
Since all of these seem to have name collisions between matrix v. vector addition, multiplication, etc. wondering if it makes sense to nest this export under some sort of class or named constant. Just want to avoid littering code with things like:

```
import { add as matrixAdd } from 'matrix3'
import { add as vectorAdd } from 'vector2'
```

as well as having to back reference whether we're operating on matrices, vectors, lists, etc. because we've lost all context of what `add` we're working with at that point...

Reply:
a module is already a logical grouping of exports. you can:

```js
import * as matrix3 from 'matrix3'
// ...
matrix3.add(a,b)
```

if ya want. also you can get context from your editor:

![image](https://user-images.githubusercontent.com/35559/72289986-41403d00-361a-11ea-88d0-8c7db0d6e980.png)